src/main/java/com/redhat/labs/lodestar/service/GitService.java

```diff
@@ -0,0 +1,337 @@
+
+		return artifacts;
+
+	}
+
+	/**
+	 * Returns a {@link List} of {@link Project} found in the configured group or
+	 * subgroups.
+	 * 
+	 * @param includeSubgroups
+	 * @return
+	 */
+	List<Project> getProjectsByGroup(Boolean includeSubgroups) {
+
+		Response response = null;
+		PagedResults<Project> page = new PagedResults<>(pageSize);
+
+		while (page.hasMore()) {
+			response = gitlabRestClient.getProjectsbyGroup(groupParentId, includeSubgroups, pageSize, page.getNumber());
+			page.update(response, new GenericType<List<Project>>() {
+			});
+		}
+
+		if (null != response) {
+			response.close();
+		}
+
+		return page.getResults();
+	}
+
+	/**
+	 * Returns a {@link List} containing the {@link ProjectTreeNode}s for the given
+	 * project ID.
+	 * 
+	 * @param projectId
+	 * @param recursive
+	 * @return
+	 */
+	List<ProjectTreeNode> getProjectTree(Integer projectId, boolean recursive) {
+
+		Response response = null;
+		PagedResults<ProjectTreeNode> page = new PagedResults<>(pageSize);
+
+		while (page.hasMore()) {
+			response = gitlabRestClient.getProjectTree(projectId, recursive);
+			page.update(response, new GenericType<List<ProjectTreeNode>>() {
+			});
+		}
+
+		if (null != response) {
+			response.close();
+		}
+
+		return page.getResults();
+
+	}
+
+	/**
+	 * Creates, updates, or deletes the {@link Artifact}s in the artifacts.json for
+	 * the given engagement ID. If any {@link Optional} parameter is not supplied
+	 * the configured default will be used.
+	 * 
+	 * @param engagementUuid
+	 * @param artifacts
+	 * @param authorEmail
+	 * @param authorName
+	 * @param commitMessage
+	 */
+	void createOrUpdateArtifactsFile(String engagementUuid, List<Artifact> artifacts, Optional<String> authorEmail,
+			Optional<String> authorName, Optional<String> commitMessage) {
+
+		// find project by engagement
+		Project project = findProjectByEngagementUuid(engagementUuid)
+				.orElseThrow(() -> new WebApplicationException("no project found with engagemnt id" + engagementUuid));
+
+		// does artifact file exist
+		boolean isUpate = artifactsFileExists(project);
+
+		// create json content
+		String content = jsonb.toJson(artifacts);
+
+		// find default branch
+		String branch = findDefaultBranch(project.getId());
+
+		// create
+		File artifactsFile = createArtifactsFile(content, branch, authorName, authorEmail, commitMessage);
+
+		// create or udpate in git
+		if (isUpate) {
+			gitlabRestClient.updateFile(project.getId(), artifactsFile.getFilePath(), artifactsFile);
+		} else {
+			gitlabRestClient.createFile(project.getId(), artifactsFile.getFilePath(), artifactsFile);
+		}
+
+	}
+
+	/**
+	 * Returns true if artifacts.json is found in the project's tree. Otherwise,
+	 * false.
+	 * 
+	 * @param project
+	 * @return
+	 */
+	boolean artifactsFileExists(Project project) {
+
+		// get project tree
+		ProjectTree projectTree = createProjectTree(project);
+
+		// update if file already exists
+		return projectTree.hasArtifactsJsonFile();
+
+	}
```

Comment:
as mentioned - should assume `true` and exception if false

Reply:
now calling git-api.